controllers/disruption_controller.go

```diff
@@ -150,6 +152,35 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{Requeue: false}, err
 		}
 
+		// handle generic safety nets if safemode is enabled
+		if !instance.Spec.Safemode.IgnoreAll {
+			// initialize all relevant safety nets for the first time
+			if len(r.SafetyNets) == 0 {
+				r.SafetyNets = []safemode.Safemode{}
+				r.SafetyNets = safemode.AddAllSafemodeObjects(*instance, r.Client)
+			}
+
+			responses := []string{}
+
+			for _, safetyNet := range r.SafetyNets {
+				// safety nets may occur throughout the reconciler, the safety nets used at creation are done here
+				response, err := safetyNet.CreationSafetyNets()
```

Comment:
I'd argue either these need to be moved into validate, or they should be checked more often than just the first Reconcile loop

Reply:
I see, this goes back to `CheckSpecSafety` as a method that checks continuously rather than just on creation. There are definitely some of the safety nets that would be futile to check more than once (`SpecificContainDisk`), but I can understand wanting to continuously check a safety net such as `CountTooLarge` because if a user has a percentage our check may change as the disruption continues. Do you think its ok for these safety nets to run more than once if they are only required once to run?

Reply:
It depends on how long they take, I think? If it's under 1-2s, I think we can just run them every loop. I'd prefer to move them into validate if they literally only make sense once at a time (e.g. specificcontaindisk)

Reply:
There are some that take a fairly long time, for example the countTooLarge makes a kube api call for many targets. Maybe this is something we can discuss as a team in regards to which one of these should move to validate.

Reply:
CountTooLarge and SporadicTargets are way too time consuming to move into the reconcile loop, imo

Reply:
I also feel that most if not all (maybe just the sporadic one but I'd still have it working in a goroutine that would itself trigger the reconcile loop if caught instead) of those safety nets could be done once during validate so we do not overload the reconcile loop. We must keep in mind that the reconcile loop is not always called and that when it is called, it blocks the controller which won't reconcile other disruptions, so better make it fast. One last argument for checking that in validate is that you can immediately reject the disruption instead of going through events.